src/main/java/com/ecore/roles/exception/InvalidArgumentException.java

```diff
@@ -1,10 +1,8 @@
 package com.ecore.roles.exception;
 
-import static java.lang.String.format;
-
 public class InvalidArgumentException extends RuntimeException {
 
-    public <T> InvalidArgumentException(Class<T> resource) {
```

Comment:
I think makes more sense use Bean Validation instead of throwing this exception. The code becomes cleaner with its annotations: @NotNull, @NotEmpty, etc...